src/opencmiss/utils/zinc/field.py

```diff
@@ -399,24 +397,159 @@ def create_field_coordinates(fieldmodule : Fieldmodule, name="coordinates", comp
+    if not nodeGroup.isValid():
+        nodeGroup = group.createFieldNodeGroup(nodeset)
+    return nodeGroup
+
+
+def create_field_texture_coordinates(fieldmodule : Fieldmodule, name="texture coordinates", components_count=3, managed=False)\
+        -> FieldFiniteElement:
+    """
+    Create texture coordinates finite element field of supplied name with
+    number of components 1, 2, or 3 and the components named "u", "v" and "w" if used.
+    New field is not managed by default.
+    """
+    return create_field_finite_element(fieldmodule, name, components_count,
+                                       component_names=("u", "v", "w"), managed=managed, type_coordinate=True)
+
+
+def find_or_create_field_texture_coordinates(fieldmodule : Fieldmodule, name="texture coordinates", components_count=3, managed=True)\
+        -> FieldFiniteElement:
+    """
+    Create texture coordinates finite element field of supplied name with
+    number of components 1, 2, or 3 and the components named "u", "v" and "w" if used.
+    New field is managed by default.
+    """
+    assert 1 <= components_count <= 3
+    return find_or_create_field_finite_element(fieldmodule, name, components_count,
+                                              component_names=("u", "v", "w"), managed=managed, type_coordinate=True)
+
+
+def create_field_stored_mesh_location(fieldmodule : Fieldmodule, mesh : Mesh, name=None, managed=False) -> FieldStoredMeshLocation:
+    """
+    Create a stored mesh location field for storing locations in the
+    supplied mesh, used for storing data projections.
+    New field is not managed by default.
+
+    :param fieldmodule:  Zinc fieldmodule to find or create field in.
+    :param mesh:  Mesh to store locations in, from same fieldmodule.
+    :param name:  Name of new field. If None, defaults to "location_" + mesh.getName().
+    :param managed: Managed state of field.
+    :return: Zinc FieldStoredMeshLocation
+    """
+    if not name:
+        name="location_" + mesh.getName()
+    with ChangeManager(fieldmodule):
+        meshLocationField = fieldmodule.createFieldStoredMeshLocation(mesh)
+        meshLocationField.setName(name)
+        meshLocationField.setManaged(managed)
+    return meshLocationField
+
+
+def find_or_create_field_stored_mesh_location(fieldmodule : Fieldmodule, mesh : Mesh, name=None, managed=True) -> FieldStoredMeshLocation:
     """
     Get or create a stored mesh location field for storing locations in the
     supplied mesh, used for storing data projections.
     Note can't currently verify existing field stores locations in the supplied mesh.
-    Asserts that existing field of name is a Stored Mesh Location field.
+    New field is managed by default.
+
     :param fieldmodule:  Zinc fieldmodule to find or create field in.
     :param mesh:  Mesh to store locations in, from same fieldmodule.
     :param name:  Name of new field. If None, defaults to "location_" + mesh.getName().
```

Comment:
Tempted to change this to `:param name:  Name of new field. If not defined defaults to "location_" + mesh.getName().`

Reply:
Fixed.